src/Model/Post/Content.php

```diff
@@ -109,16 +109,15 @@ public static function delete(array $conditions, array $options = [])
 	public static function getURIIdListBySearch(string $search, int $uid = 0, int $start = 0, int $limit = 100, int $last_uriid = 0)
 	{
 		$condition = ["`uri-id` IN (SELECT `uri-id` FROM `post-content` WHERE MATCH (`title`, `content-warning`, `body`) AGAINST (? IN BOOLEAN MODE))
-			AND (NOT `private` OR (`private` AND `uid` = ?)) AND (`network` IN (?, ?, ?, ?) OR (`uid` = ? AND `uid` != ?))",
-			$search, $uid, Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS, $uid, 0];
+			AND (`uid` = ? OR (`uid` = ? AND NOT `global`)) AND (`network` IN (?, ?, ?, ?) OR (`uid` = ? AND `uid` != ?))",
```

Comment:
``(`uid` = ? OR (`uid` = ? AND NOT `global`))`` can be simplified to `` `uid` = ? `` as the value of `` `global` `` is irrelevant in this condition. However, the new condition isn't similar to the old one with `` `private` ``. Are you sure that both ``NOT `private` `` and `` `private` `` are supposed to be replaced by `` `uid` = ? `` ?

Reply:
No. The first `uid` is set to `0`, the second one to the `uid` of the current user. And `global` is set on each post that has got a `post-user` entry with an `uid` of `0`. Technically there can be global posts that don't have a `uid=0` entry. These are - for example - posts from connectors.

Reply:
Ah, sorry, I didn't check the values that are supposed to be interpolated in the `?`, thanks for the elaboration.

Reply:
Yeah, even though the first value is a constant, I prefer to set it like a variable. I think this is cleaner.